sklearn/utils/stats.py

```diff
@@ -29,41 +28,60 @@ def _weighted_percentile(array, sample_weight, percentile=50):
     percentile : int if `array` 1D, ndarray if `array` 2D
         Weighted percentile.
     """
+    xp, _ = get_namespace(array)
     n_dim = array.ndim
     if n_dim == 0:
         return array[()]
```

Comment:
I think this code will keep working and do the right thing for array API inputs. At least the following works:
```python
In [12]: x_ = array_api_strict.asarray(1)

In [13]: x_.ndim
Out[13]: 0

In [14]: x[()]
Out[14]: np.int64(1)
```
(but maybe this only works because array api strict uses numpy in the background?

Reply:
Thank you @betatim.

I have a question: what do you mean when you say, the code works? I am not sure what the distinction criterium for working and not working is, since we won't get any error either way. Do we need to measure performance?
If you are looking for the data type: Why did you expect `np.int64` here?
(And since I am lacking some knowledge: if you could recommend some terms I can google to find out about data types and their storage in memory, that would be highly appreciated, too.)